trackintel/preprocessing/positionfixes.py

```diff
@@ -173,187 +177,335 @@ def generate_triplegs(positionfixes, staypoints=None, *args, **kwargs):
+        return ret_pfs, ret_tpls
+
+    else:
+        raise NameError('Chosen method is not defined')
+
+
+def _triplegs_between_staypoints_case1(positionfixes, staypoints, user_id_this):
+    """
+    This function uses the staypoints and the column 'staypoint_id' in the positionfixes, to identify all
+    positionfixes that lie in between two staypoints.
+
+    Parameters
+    ----------
+    positionfixes: trackintel positionfixes
+    staypoints: trackintel staypoints
+    user_id_this:
+
+    Returns
+    --------
+    list
+        a list of dictionaries with individual triplegs
+    """
+
+    generated_triplegs_list = []
+    spts = staypoints.loc[staypoints['user_id'] == user_id_this].sort_values('started_at')
+    spts = spts.reset_index().to_dict('records')
+
+    for spt1, spt2 in zip(list(spts), list(spts)[1:]):
+        # - Go through all pairs of consecutive staypoints.
+        # - identify end of first and start of second staypoint.
+        # - assign all positionfixes in between (including bounds) to a tripleg
+
+        # get the last posfix of the first staypoint
+        index_first_posfix_tl = positionfixes[positionfixes.staypoint_id == spt1['id']].index[-1]
+        position_first_posfix_tl = positionfixes.index.get_loc(index_first_posfix_tl)
+
+        # get first posfix of the second staypoint
+        index_last_posfix_tl = positionfixes[positionfixes.staypoint_id == spt2['id']].index[0]
+        position_last_posfix_tl = positionfixes.index.get_loc(index_last_posfix_tl)
+
+        # create tripleg from all positionfixes in between the two staypoints
+        pfs_tripleg = positionfixes.iloc[position_first_posfix_tl:position_last_posfix_tl + 1]
+        generated_triplegs_list.append(_get_tripleg_record_from_psfs(pfs_tripleg, user_id_this, min_nb_of_points=3))
+
+    # add first tripleg to the beginning of generated_tripleg_list
+    index_first_posfix_first_stp = positionfixes[positionfixes.staypoint_id == spts[0]['id']].index[0]
+    position_first_posfix_first_stp = positionfixes.index.get_loc(index_first_posfix_first_stp)
+
+    pfs_tripleg = positionfixes.iloc[0:position_first_posfix_first_stp + 1]
+    generated_triplegs_list = [_get_tripleg_record_from_psfs(pfs_tripleg, user_id_this, min_nb_of_points=2)] + \
+                              generated_triplegs_list
+
+    # add last tripleg to the end of generated_triplegs
+    index_last_posfix_last_stp = positionfixes[positionfixes.staypoint_id == spts[-1]['id']].index[-1]
+    position_last_posfix_last_stp = positionfixes.index.get_loc(index_last_posfix_last_stp)
+
+    pfs_tripleg = positionfixes.iloc[position_last_posfix_last_stp:]
+    generated_triplegs_list.append(_get_tripleg_record_from_psfs(pfs_tripleg, user_id_this, min_nb_of_points=2))
+
+    # filter None values
+    return list(filter(None, generated_triplegs_list))
+
+
+def _get_tripleg_record_from_psfs(pfs_tripleg, user_id_this, min_nb_of_points):
```

Comment:
This function is used by both case1 and case2? Would be nice to place it after the two functions, and maybe add two `_` (i.e., `__`) as prefix.

Reply:
👍